packages/slider/index.tsx

```diff
@@ -0,0 +1,380 @@
+        evtType: EventType
+      ): void => {
+        if (evtType === this.transitionendEvtName) {
+          this.onThumbEnd = undefined;
+        }
+      },
+      registerBodyInteractionHandler: (
+        evtType: EventType,
+        handler: SpecificEventListener<EventType>
+      ): void => {
+        document.body.addEventListener(evtType, handler);
+      },
+      deregisterBodyInteractionHandler: (
+        evtType: EventType,
+        handler: SpecificEventListener<EventType>
+      ): void => {
+        document.body.removeEventListener(evtType, handler);
+      },
+      registerResizeHandler: (
+        handler: SpecificEventListener<'resize'>
+      ): void => {
+        window.addEventListener('resize', handler);
+      },
+      deregisterResizeHandler: (
+        handler: SpecificEventListener<'resize'>
+      ): void => {
+        window.removeEventListener('resize', handler);
+      },
+      notifyInput: (): void => {
+        const callback = this.props.onValueInput;
+        if (typeof callback === 'function') {
+          callback(this.foundation.getValue());
+        }
+      },
+      notifyChange: (): void => {
+        const callback = this.props.onValueChange;
+        if (typeof callback === 'function') {
+          callback(this.foundation.getValue());
+        }
+      },
+      setThumbContainerStyleProperty: (prop: string, value: string): void =>
+        this.setStyleToElement(prop, value, 'thumbContainerStyleProperty'),
+      setTrackStyleProperty: (prop: string, value: string): void =>
+        this.setStyleToElement(prop, value, 'trackStyleProperty'),
+      setMarkerValue: (value: number): void =>
+        this.setState({markerValue: value}),
+      appendTrackMarkers: (numMarkers: number): void => {
+        this.setState((prevState) => {
+          let {markerStyles} = prevState;
+          // Shallow copy of array:
+          markerStyles = markerStyles.slice();
+          for (let i = 0; i < numMarkers; i++) {
+            markerStyles.push({});
+          }
+          return {markerStyles};
+        });
+      },
+      removeTrackMarkers: (): void => {
+        this.setState({markerStyles: []});
+      },
+      setLastTrackMarkersStyleProperty: (
+        propertyName: string,
+        value: string
+      ): void => {
+        this.setState((prevState) => {
+          let {markerStyles} = prevState;
+          const last = markerStyles.length - 1;
+          markerStyles = markerStyles.slice();
+          const clone = Object.assign({}, markerStyles[last]);
+          // we need to cast prop from string (interface requirement) to React.CSSProperties;
+          const typedProp = propertyName as keyof React.CSSProperties;
+          // https://github.com/Microsoft/TypeScript/issues/11914
+          clone[typedProp] = value;
+          markerStyles[last] = clone;
+          return {markerStyles};
+        });
+      },
+      isRTL: (): boolean => this.props.isRtl || false,
```

Comment:
|| false is unnecessary as you have it defined in defaultProps.

Reply:
Removed optional types for props per https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#support-for-defaultprops-in-jsx